dec/decode.go

```diff
@@ -77,14 +77,20 @@ type BrotliReader struct {
 	// Internal buffer for compressed data
 	buffer []byte
 
-	availableIn C.size_t
-	nextIn      *C.uint8_t
-	totalOut    C.size_t
+	availableIn     C.size_t
+	nextIn          *C.uint8_t
+	totalOut        C.size_t
+	eof             bool
```

Comment:
There might be a way to make it all work without having an `eof` marker in the struct, by reworking the structure of `Read` a little.. (if we read nothing so far AND last read gave io.EOF AND we didn't need more output, then return eof) - but not sure it's worth bothering for a bool, not sure which is the most readable version, at any rate this one seems correct